impl-base/src/main/java/org/jboss/shrinkwrap/impl/base/importer/zip/ZipImporterImpl.java

```diff
@@ -214,10 +214,16 @@ private ZipImporter importFrom(final ZipFile file, Filter<ArchivePath> filter) t
                     continue;
                 }
 
-                archive.add(new ZipFileEntryAsset(file, entry), new BasicPath(entryName));
+                archive.add(new ZipFileEntryAsset(new File(file.getName()), entry), new BasicPath(entryName));
```

Comment:
The ZipFileEntryAsset now just gets a simple java.io.File and ZipImporterImpl closes the original ZipFile (which was used for iteration and filtering) - This releases the Windows file handle.

A call to ZipFileEntryAsset#openStream() now uses the java.io.File to create a new ZipFile and stream, which is wrapped. Closing this wrapped stream now also closes the ZipFile, again releasing the Windows handle. So each call to ZipFileEntryAsset#openStream() repeats this, each time ensuring the Windows handle is released safely.